### ISSUE TITLE
Clipboard not cleared when copying passwords from "Entry > Advanced > Additional attributes > [value of attribute]"

### ISSUE THREAD

**Issue by notDavid** [keepassxreboot/keepassxc#10524]
## Overview
Hi there, not sure if it's a bug, or a feature request... please advise thank you!

## Steps to Reproduce
1. Enable clipboard clearing after X seconds
2. Create an entry, generate a password, and save it in an "Advanced Attribute" field
3. Click "Protect" and "Reveal"
4. Select the password from the "Advanced Attribute" value and copy the password to the clipboard with `Cmd + v`
5. The clipboard is never cleared (i think it's also not marked as sensitive / a password, in the clipboard, either)

## Expected Behavior
To clear the clipboard after X seconds

## Actual Behavior
The clipboard is never cleared

[NOTE]: # ( Paste debug info from Help → About here )
KeePassXC - Version 2.7.7
Revision: 68e2dd8

Qt 5.15.11
Debugging mode is disabled.

Operating system: macOS 14.4
CPU architecture: arm64
Kernel: darwin 23.4.0

Enabled extensions:
- Auto-Type
- Browser Integration
- Passkeys
- SSH Agent
- KeeShare
- YubiKey
- Quick Unlock

Cryptographic libraries:
- Botan 3.1.1

[NOTE]: # ( Pick choices based on your environment )
Operating System: macOS

**Comment by notDavid** [comment]
PS. Note that, the password *is cleared correctly*, after using the "Password Generator", which i mentioned in Step 2 above... but **not** when copying to the clipboard from an "Advanced Attribute".

**Comment by droidmonkey** [comment]
copying directly from a text edit field does not invoke the clipboard security since that process is handled directly by Qt. If you want to copy advanced attributes to clipboard with clearing use the right-click context menu from the main entry view.

![image](https://github.com/keepassxreboot/keepassxc/assets/2809491/433618d2-6456-403b-a6d9-409fc4071a46)

**Comment by notDavid** [comment]
Thank you @droidmonkey , good to know there's an alternative.

I doubt whether the average users will be aware of this, it seems the most intuitive way to find and copy these passwords. So im curious, is it not implemented due to technical difficulties, or...?

Would it be a good idea to consider implementing, to prevent "leaks" ? (i only see advantages.)

Please consider, or feel free to close this issue otherwise. Thank you!

**Comment by kaoneko** [comment]
I'd say that right click menu is _exactly_ where most users will look. Or the preview panel, where you can double click on an attribute's value to copy it (even when it's obscured). "Editing" an entry just to copy one of its secrets? That just seems weird to me 😄 And a very complicated way to achieve your goal ⛳️

**Comment by notDavid** [comment]
> I'd say that right click menu is exactly where most users will look.

Yes, if you know what you're after. But not if you need to get an overview first of which items are in there.

>  Or the preview panel, where you can double click on an attribute's value to copy it

I didn't even know that! ☺️ (and i've been using KeePassXC for many years.) It's sort of, a hidden feature... 🤷🏼‍♂️

Good to know, though.

**Comment by FlippingBinary** [comment]
The ctrl-c shortcut really should trigger clipboard clearing after copying text from a protected attribute, IMHO, and it wouldn't be a bad thing if it started the countdown after copying any text -- even unprotected text in the search field. 

The thing that was really unexpected for me (and the reason I found this issue) is that when you select the name of a protected attribute in the advanced tab of an open entry, ctrl-c copies the entry's password instead of the value of the protected attribute. The "Copy password" menu item under the "Entries" menu is enabled and has the ctrl-c shortcut listed next to it, so my surprise could be chalked up to user error. But that menu item is still enabled with the same shortcut listed even when the cursor is in a text field where ctrl-c will not copy the password.

Perhaps the shortcuts (and their corresponding menu items) that copy pre-defined fields from an entry should be disabled when an entry is opened for editing so that ctrl-c could consistently copy the value of whatever is selected (or highlighted text) with regular clipboard-clearing behavior. If nothing is selected or highlighted, nothing gets copied (but maybe a toast message could remind the user about ways to copy the entry's password?).

Obviously, there aren't any surprises once you know the quirks of how it works, so I'm all set, but new users might benefit from those tweaks.

Great app, though. It's been rock solid for me. I just can't believe it didn't cross my mind to use the right-click menu to copy the value of an attribute, lol. It's like my mind blocked it out as a possibility because there's no keyboard shortcut for it. That double-click trick on the attributes in the preview panel is nice too. Thanks, @droidmonkey and @kaoneko!

**Comment by droidmonkey** [comment]
If you are on macos, then the copy password unexpected behavior is a qt bug that we have a fix in for next release.

**Comment by K0JIbKA** [comment]
Same issue with clipboard clearing. 
But in my case none of the clipboards are cleaned at all.
Whenever I do 'Ctrl+c' or copy password from context menu, or copy any attribute, or copy newly generated password from Generator - none of it gets cleared from the clipboard and remains there constant.
Not sure if it should be reported as a new bug, so I just leave it here.

I have following short description of my system:
KeePassXC - Version 2.7.10
Revision: b342be4

Qt 5.15.15
Debugging mode is disabled.

Operating system: Debian GNU/Linux 13 (trixie)
CPU architecture: x86_64
Kernel: linux 6.12.95+deb13-amd64

Enabled extensions:
- Auto-Type
- Browser Integration
- Passkeys
- SSH Agent
- KeeShare
- YubiKey
- Secret Service Integration

Cryptographic libraries:
- Botan 2.19.5

**Comment by droidmonkey** [comment]
If you are using wayland and gnome then you need to install wl-clipboard
